app/storage.js

```diff
@@ -0,0 +1,103 @@
+const { DefaultAzureCredential } = require('@azure/identity')
+const { BlobServiceClient } = require('@azure/storage-blob')
+const config = require('./config/storage')
+let blobServiceClient
+let containersInitialised
+let foldersInitialised
+
+if (config.useConnectionStr) {
+  console.log('Using connection string for BlobServiceClient')
+  blobServiceClient = BlobServiceClient.fromConnectionString(config.connectionStr)
+} else {
+  console.log('Using DefaultAzureCredential for BlobServiceClient')
+  const uri = `https://${config.storageAccount}.blob.core.windows.net`
+  blobServiceClient = new BlobServiceClient(uri, new DefaultAzureCredential())
+}
+
+const container = blobServiceClient.getContainerClient(config.container)
+
+const initialiseContainers = async () => {
+  if (config.createContainers) {
+    console.log('Making sure blob containers exist')
+    await container.createIfNotExists()
+    console.log('Containers ready')
+  }
+  foldersInitialised ?? await initialiseFolders()
+  containersInitialised = true
+}
+
+const initialiseFolders = async () => {
+  console.log('Making sure folders exist')
+  const placeHolderText = 'Placeholder'
+  const inboundClient = container.getBlockBlobClient(`${config.inboundFolder}/default.txt`)
+  await inboundClient.upload(placeHolderText, placeHolderText.length)
+  foldersInitialised = true
+  console.log('Folders ready')
+}
+
+const getBlob = async (folder, filename) => {
+  containersInitialised ?? await initialiseContainers()
+  return container.getBlockBlobClient(`${folder}/${filename}`)
+}
+
+const getFile = async (filename) => {
+  filename = sanitizeFilename(filename)
+  console.log(`Searching for ${filename}`)
+  const blob = await getBlob(config.inboundFolder, filename)
+  const downloaded = await blob.downloadToBuffer()
+  console.log(`Found ${filename}`)
+  return downloaded.toString()
+}
+
+const getInboundFileList = async () => {
```

Comment:
I dnt think this name is v clear.

I thought it would return all files in the `inbound` folder but there's filtering going on

Reply:
I can see in `verify/index.js` the output from this is referred to as `batchControlFilename`

Reply:
Agreed.  renamed to `getPendingControlFiles` as that's what it's retrieving.  It effectively replaces the Function trigger.